File: app/src/main/java/com/rum/myapplication/generalHelper/SP.java
package com.rum.myapplication.generalHelper;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

/**
 * Created by nectarbits on 15/02/16.
 */
public class SP {

    public static final String SP_TRUE = "SP_TRUE";
    public static final String SP_FALSE = "SP_FALSE";
    public static final String LOGIN_STATUS = "LOGIN_STATUS";
    public static final String USER_FIRST_NAME = "USER_FIRST_NAME";
    public static final String USER_LAST_NAME = "USER_LAST_NAME";
    public static final String USER_MOBILE = "USER_MOBILE";
    public static final String USER_EMAIL = "USER_EMAIL";
    public static final String USER_ID = "USER_ID";
    public static final String USER_LAT = "USER_LAT";
    public static final String USER_LONG = "USER_LONG";
    public static final String SIGNATURE_IMAGE_PATH = "SIGNATURE_IMAGE_PATH";
    public static final String ON_DUTY = "ON_DUTY";

    /**
     * @param mContext
     * @param key
     * @param value
     */
    public static void savePreferences(Context mContext, String key, String value) {
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(mContext);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(key, value).apply();
    }


    /**
     * @param context
     * @param keyValue
     * @return
     */
    public static String getPreferences(Context context, String keyValue) {
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        return sharedPreferences.getString(keyValue, "");
    }

    /**
     * @param mContext
     */
    public static void removeAllSharedPreferences(Context mContext) {
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(mContext);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.clear().apply();
    }
}

File: app/src/main/java/com/rum/myapplication/utils/SP.kt
package com.rum.myapplication.utils

import android.content.Context
import android.preference.PreferenceManager

object SP {

    const val SELECTED_LANGUAGE = "SELECTED_LANGUAGE"
    const val LANGUAGE_ENGLISH = "en"
    const val LANGUAGE_HINDI = "hi"
    const val LANGUAGE_RUSSIAN = "ru"
    const val LANGUAGE_UZBEK = "uz"
}

fun Context.getPreferences(keyValue: String): String? {
    val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this)
    return sharedPreferences.getString(keyValue, "")
}

fun Context.removeAllSharedPreferences() {
    val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this)
    val editor = sharedPreferences.edit()
    editor.clear().apply()
}

fun Context.savePreferences(key: String, value: String?) {
    val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this)
    val editor = sharedPreferences.edit()
    editor.putString(key, value).apply()
}